components/vocabularies/hgnc/ui/src/main/resources/PhenoTips/GeneSymbolDisplayer.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version='1.0' encoding='UTF-8'?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+-->
+
+<xwikidoc version="1.1">
+  <web>PhenoTips</web>
+  <name>GeneSymbolDisplayer</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <creationDate>1476713513000</creationDate>
+  <parent>PhenoTips.PatientSheet</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <date>1476716507000</date>
+  <contentUpdateDate>1476716507000</contentUpdateDate>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#macro (_displayGeneSymbol $prefix $name $value)
+  #set ($termStatus = "")
+  #set ($gene = {'id' : $!{escapetool.xml($value)}})
+  #set ($term = $services.vocabularies.hgnc.getTerm("${gene.id}"))
+  #if ($term)
+    #set ($gene.symbol = $term.symbol)
+  #else
+    #_validateGeneId($gene, $termStatus)
+    #set ($gene.symbol = $gene.id)
+  #end
+  #set ($placeholder = $services.localization.render('Phenotips.GeneClass.geneNamePlaceholder'))
+  #if ($xcontext.action == 'edit')
+    {{html wiki="false" clean="false"}}&lt;input type='text' name='${prefix}symbol' class='suggested suggest-gene gene-name $!termStatus' placeholder='$placeholder' size='16' value='$!{gene.symbol}'/&gt;&lt;input type='hidden' name='${prefix}${name}' class='gene-id' value='$!{gene.id}'/&gt;{{/html}}
+  #else
+    ${gene.symbol}#if("$!{gene.symbol}" != ''){{html wiki="false" clean="false"}}&lt;input type='hidden' name='gene_name' class='gene-name' value="$!{gene.symbol}"/&gt;{{/html}}#end
+  #end
+#end
+##
+##
+#macro (_validateGeneId $geneData $termStatus)
+  #if (!$geneData.id.matches('^ENSG[0-9]{11}$'))
+    #set ($termStatus = "nonstandard")
```

Comment:
I overheard Alina and Orion talking about this, and how anything that matches the Ensembl id should be accepted, was it was because HGNC is not always updated? @buske please jump in with more details.